idaes/generic_models/properties/core/eos/virial.py

```diff
@@ -0,0 +1,797 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Methods for virial equations of state (veos) for gases
+
+Currently supports B-truncated volume explicit virial equation of state with validity at low to moderate pressures where the compresiibility factor is approximately a linear function of  pressure. It is most accurate for non-polar species.
+
+Refernces**
+[1] J. M Prausnitz, R, N Lichtenthaler, and E. G de Azvedo,
+    Molecular thermodynamics of Fluid-phase Equilibruim 3rd ed. Prentice-Hall, Nj, 1998
+**Any standard chemical engineering thermodynamic textbook
+
+Author: Akula Paul
+
+"""
+
+from copy import deepcopy
+from pyomo.environ import (Expression,
+                           log, exp,
+                           value, NonNegativeReals,
+                           Var, units as pyunits)
+from pyomo.common.config import ConfigBlock, ConfigValue, Bool
+
+from idaes.core.util.exceptions import PropertyNotSupportedError
+from idaes.generic_models.properties.core.generic.utility import (
+    get_method, get_component_object as gcobj)
+from idaes.generic_models.properties.core.eos.eos_base import EoSBase
+from idaes.core.util.misc import set_param_from_config
+from idaes.generic_models.properties.core.pure.RPP4 import entr_mol_ig_comp
+
+
+VirialConfig = ConfigBlock()
+
+# virial eos options
+VirialConfig.declare("use_pseudocritical_rules", ConfigValue(
+    default=True,
+    domain=Bool,
+    description='''Flag indicating whether pseudocritical rules should be used
+    when applicable''',
+    doc='''Flag indicating whether to compute mixture property based on
+    composition and critical properties of pure components:
+**default** - True.
+**Valid values:** {
+**True** - use pseudocritical_rule: mixing rule based on composition and
+critical properties of pure components.
+**False** - use mixing rule based on composition and corresponding
+pure component properties.}'''))
+
+
+class enth_mol_ig_comp():
```

Comment:
If there is a bug in a separate module, you should be fixing it there, not recreating it. Please open a separate PR for this.